codemeta/parsers/gitlab.py

```diff
@@ -0,0 +1,137 @@
+import sys
+import requests
+import time
+from os import environ
+from datetime import datetime
+from io import StringIO
+from typing import Union, IO
+from rdflib import Graph, URIRef, BNode, Literal
+from rdflib.namespace import RDF
+from codemeta.common import AttribDict, SDO, CODEMETA, license_to_spdx, parse_human_name, generate_uri
+from codemeta.parsers.jsonld import parse_jsonld_data
+
+gitlab_crosswalk_table = {
+    SDO.codeRepository: "web_url",
+    SDO.dateCreated: "created_at",
+    SDO.dateModified: "last_activity_at",
+    SDO.description: "description",
+    SDO.name: "name",
+    SDO.WebPage: "web_url"
```

Comment:
Hello, 
tried to change SDO.WebPage in SDO.url but I got this problem

[harvester info] Getting contributors from git...
[harvester info] Extracting last and first commit date from git log....
[harvester info] Querying Github/GitLab API (https://gitlab.com/MartinDoskar/modular-topology-optimization)
-- begin log --
Passed 1 files/sources but specified 0 input types! Automatically guessing types...
Detected input types: [('https://gitlab.com/MartinDoskar/modular-topology-optimization', 'web')]
URI automatically generated, may be overriden later: http://10.160.0.24:80/modular-topology-optimization
Processing source #1 of 1
Obtaining metadata from remote URL https://gitlab.com/MartinDoskar/modular-topology-optimization
Traceback (most recent call last):
  File "/usr/local/bin/codemetapy", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/codemeta/codemeta.py", line 128, in main
    output = build(**args.__dict__)
  File "/usr/local/lib/python3.9/dist-packages/codemeta/codemeta.py", line 285, in build
    for targetres in codemeta.parsers.web.parse_web(g, res, source, args):
  File "/usr/local/lib/python3.9/dist-packages/codemeta/parsers/web.py", line 65, in parse_web
    r.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://gitlab.com/MartinDoskar/modular-topology-optimization
-- end log --
[harvester error] conversion from Github/GitLab API query failed for my-yaml (https://gitlab.com/MartinDoskar/modular-topology-optimization) (codemetapy failed)
[harvester info] Found releaseNotes

Reply:
Hmm, I'm investigating... Something indeed goes wrong there. It's auto-detecting input type 'web' but it should be 'gitlab' (and even then I wonder why you run into a 404 later on, it is a valid URL after all). I did find you introduced a small error in the auto-detect code (see my other comment), but I can't imagine that to have been the cause of this.

Reply:
Ah, I get it now. It's because codemeta-harvester's Dockerfile grabs codemetapy from my master branch, which does not have your changes merged yet (so it doesn't know about gitlab at all yet).

Reply:
I just pushed a commit that should remedy this (you'll need to merge it into your branch): https://github.com/proycon/codemeta-harvester/commit/95cfc6542621676384056222b6902018085cad3a

You can then at build time build against your own codemetapy: `docker build --build-arg=CODEMETAPY_VERSION=git+https://github.com/xmichele/codemetapy.git@master`. If not set, it will grab the latest stable codemetapy from pypi instead.